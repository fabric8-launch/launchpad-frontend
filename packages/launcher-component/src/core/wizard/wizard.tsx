import { Button, ButtonVariant } from '@patternfly/react-core';
import { ArrowAltCircleRightIcon, OutlinedCheckCircleIcon, LockIcon } from '@patternfly/react-icons';
import classNames from 'classnames';
import * as React from 'react';
import './wizard.scss';

export function WizardStepFooter(props: React.PropsWithChildren<{}>) {
  return (
    <div className="wizard-step-footer">
      {props.children}
    </div>
  );
}

interface WizardButtonProps {
  title?: string;
  onClick?: () => void;
  disabled?: boolean;
}

export function WizardButton({ title, onClick, disabled = false }: WizardButtonProps) {
  const text = title || 'Next';
  return (
    <Button className={'wizard-button'} aria-label={`${text} button`} onClick={onClick} isDisabled={disabled} variant={ButtonVariant.secondary}>
      <ArrowAltCircleRightIcon /> {text}
    </Button>
  );
};

interface WizardStepProps {
  title: string;
  completed?: boolean;
  summary?: string;
  selected?: boolean;
  locked?: boolean;
  onClick?: () => void;
}

export function WizardStep(props: React.PropsWithChildren<WizardStepProps>) {
  const { summary, title, completed = false, selected = false, locked = false, onClick, children } = props;
  const linkOnClick = locked || !onClick ? () => { } : () => onClick();
  const completedAndNotLocked = !locked && completed;
  let icon = (<OutlinedCheckCircleIcon style={{color: '#5cb85c'}} />);
  if(locked) {
    icon = (<LockIcon style={{color: '#bdc2ce'}}/>)
  }

  return (
    <li className={classNames({ completed: completedAndNotLocked, selected, locked })}>
      <a href="#" onClick={linkOnClick}>{title}
        {(completedAndNotLocked || locked) && icon}
      </a>
      {!locked && completed && !selected && summary && (
        <span className="summary">{summary}</span>
      )}
      {!locked && selected && (
        <span className="content">{children}</span>
      )}
    </li>
  );
}

export function Wizard(props: React.PropsWithChildren<{}>) {
  return (
    <div className="wizard">
      <ul>
        {props.children}
      </ul>
    </div>
  );
}
