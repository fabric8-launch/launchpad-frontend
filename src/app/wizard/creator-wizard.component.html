<div class="container-fluid">
  <f8launcher (onCancel)="back()" [gettingStartedInfo]="true">
    <div class="f8launcher-container_nav">
      <f8launcher-step-indicator #stepIndicator></f8launcher-step-indicator>
    </div>
    <div class="f8launcher-container_main">
      <f8launcher-targetenvironment-step #targetEnvStep [hidden]="false" id="TargetEnvironment" styleClass="target-environment"
        title="Select Target Environment">
      </f8launcher-targetenvironment-step>
      <f8launcher-runtime-step id="Runtimes" title="Select a Runtime"></f8launcher-runtime-step>
      <f8launcher-capabilities-step id="Capabilities" title="Select a Capability"></f8launcher-capabilities-step>
      <f8launcher-gitprovider-step #gitProvider [hidden]="!(targetEnvStep.selection.dependencyCheck.targetEnvironment === 'os' && targetEnvStep.completed)"
        id="GitProvider" styleClass="git-provider" title="Authorize Git Provider">
      </f8launcher-gitprovider-step>
      <f8launcher-projectsummary-step id="ProjectSummary" styleClass="project-summary" title="Confirm Application Summary & Setup">
        <f8launcher-runtime-review stepId="Runtimes"></f8launcher-runtime-review>
        <f8launcher-capabilities-review stepId="Capabilities"></f8launcher-capabilities-review>
        <f8launcher-target-environment-review stepId="shared"></f8launcher-target-environment-review>
        <f8launcher-gitprovider-review [stepId]="gitProvider.id" *ngIf="targetEnvStep.selection.dependencyCheck.targetEnvironment === 'os' && targetEnvStep.completed"></f8launcher-gitprovider-review>
      </f8launcher-projectsummary-step>
    </div>
    <div class="f8launcher-container_completed-footer">
      <button class="btn btn-primary btn-lg" (click)="back()">
        View New Application
      </button>
    </div>
  </f8launcher>
</div>