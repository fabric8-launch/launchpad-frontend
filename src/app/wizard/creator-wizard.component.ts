import { Component } from '@angular/core';
import { Router } from '@angular/router';

@Component({
  selector: 'creator-wizard',
  templateUrl: './creator-wizard.component.html'
})
export class CreatorWizardComponent {

  constructor(private router: Router) {
  }

  public back(): void {
    this.router.navigate(['/']);
  }
}
