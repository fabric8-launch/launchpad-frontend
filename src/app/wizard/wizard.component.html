<div authentication="true">
  <div class="blank-slate-pf ">
    <h1>
      To continue, please log into or register an account for free
      <br /> with the Red Hat Developer Program.
    </h1>
    <div class="blank-slate-pf-main-action">
      <button class="btn btn-primary btn-lg" (click)="authService.login()">Log in or register</button>
    </div>
  </div>
</div>
<div class="container-fluid" authentication>
  <f8launcher (onCancel)="back()" [gettingStartedInfo]="true">
    <div class="f8launcher-container_nav">
      <f8launcher-step-indicator #stepIndicator></f8launcher-step-indicator>
    </div>
    <div class="f8launcher-container_main">
      <f8launcher-targetenvironment-step #targetEnvStep [hidden]="isHidden()" id="TargetEnvironment" styleClass="target-environment"
        title="Target Environment">
      </f8launcher-targetenvironment-step>
      <flow-choice id="FlowChoice" title="Example or Create New" #flowStep [hidden]="!creatorEnabled || (targetEnvStep.selection.cluster && targetEnvStep.selection.cluster.type === 'starter')"></flow-choice>
      <missionruntime-step #missionRuntimeStep id="MissionRuntime" styleClass="mission-runtime"
        title="Mission & Runtime" [canChangeVersion]="true" [hidden]="flowStep.selection.creatorFlow">
      </missionruntime-step>
      <f8launcher-runtime-step id="Runtimes" title="Runtime" [hidden]="!flowStep.selection.creatorFlow"></f8launcher-runtime-step>
      <f8launcher-capabilities-step id="Capabilities" title="Capabilities" [hidden]="!flowStep.selection.creatorFlow"></f8launcher-capabilities-step>
      <f8launcher-gitprovider-step #gitProvider [hidden]="!(targetEnvStep.selection.dependencyCheck.targetEnvironment === 'os' && targetEnvStep.completed)"
        id="GitProvider" styleClass="git-provider" title="Git Provider">
      </f8launcher-gitprovider-step>
      <f8launcher-projectsummary-step id="ProjectSummary" styleClass="project-summary" title="Application Summary">
        <f8launcher-target-environment-review stepId="shared"></f8launcher-target-environment-review>
        <f8launcher-gitprovider-review [stepId]="gitProvider.id" *ngIf="targetEnvStep.selection.dependencyCheck.targetEnvironment === 'os' && targetEnvStep.completed"></f8launcher-gitprovider-review>
        <f8launcher-runtime-review stepId="Runtimes" *ngIf="flowStep.selection.creatorFlow"></f8launcher-runtime-review>
        <f8launcher-capabilities-review stepId="Capabilities" *ngIf="flowStep.selection.creatorFlow"></f8launcher-capabilities-review>
        <f8launcher-mission-runtime-review [stepId]="missionRuntimeStep.id" *ngIf="!flowStep.selection.creatorFlow"></f8launcher-mission-runtime-review>
      </f8launcher-projectsummary-step>
    </div>
    <div class="f8launcher-container_completed-footer">
      <button class="btn btn-primary btn-lg" (click)="back()">
        Start a new project
      </button>
    </div>
  </f8launcher>
</div>