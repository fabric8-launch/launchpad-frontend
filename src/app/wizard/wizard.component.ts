import { Component } from '@angular/core';
import { Router } from '@angular/router';
import { Config } from 'ngx-launcher';
import { AuthService } from 'app/shared/auth.service';

@Component({
  selector: 'wizard',
  templateUrl: './wizard.component.html'
})
export class WizardComponent {

  constructor(private router: Router, private config: Config,
              public authService: AuthService) {
  }

  public back(): void {
    this.router.navigate(['/']);
  }

  public isHidden(): boolean {
    return this.config.get('targetenvironment_skip') === 'true';
  }

  public get creatorEnabled(): boolean {
    return this.config.get('creator_enabled') === 'true';
  }
}
