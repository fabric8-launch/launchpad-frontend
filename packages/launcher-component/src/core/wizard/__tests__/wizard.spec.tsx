import * as React from 'react';
import 'jest-dom/extend-expect';
import { Wizard, WizardStep, WizardButton } from '../wizard';
import { render, fireEvent, cleanup } from 'react-testing-library';

afterEach(() => cleanup());

describe('<Wizard />', () => {
  it('renders the wizard with one step', () => {
    const component = render(<Wizard><WizardStep title={'my first step'} selected={true} locked={false} /></Wizard>);
    expect(component).toMatchSnapshot();
  });
});

describe('<Wizard.Step />', () => {
  it('renders the step as completed', () => {
    const component = render(<WizardStep title={'my first step'} completed={true} />);
    expect(component).toMatchSnapshot();
  });
  it('renders the step as completed and locked', () => {
    const component = render(<WizardStep title={'my first step'} completed={true} locked={true} />);
    expect(component).toMatchSnapshot();
  });

});

describe('<Wizard.Button />', () => {
  it('renders the button with type next', () => {
    const component = render(<WizardButton title={'Go to next'} />);
    expect(component).toMatchSnapshot();
    expect(component.getByLabelText('Go to next button')).toBeDefined();
  });

  it('renders the button with type launch', () => {
    const component = render(<WizardButton />);
    expect(component).toMatchSnapshot();
    expect(component.getByLabelText('Next button')).toBeDefined();
  });

  it('renders the button disabled', () => {
    const component = render(<WizardButton disabled={true} />);
    expect(component).toMatchSnapshot();
    expect(component.getByLabelText('Next button')).toHaveAttribute('disabled');
  });

  it('call function when clicking', () => {
    const mockCallBack = jest.fn();
    const component = render(<WizardButton onClick={mockCallBack} />);
    fireEvent.click(component.getByLabelText('Next button'));
    expect(mockCallBack).toHaveBeenCalled();
  });
});