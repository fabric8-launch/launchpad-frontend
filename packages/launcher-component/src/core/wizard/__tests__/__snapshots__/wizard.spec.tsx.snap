// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wizard /> renders the wizard with one step 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wizard"
      >
        <ul>
          <li
            class="selected"
          >
            <a
              href="#"
            >
              my first step
            </a>
            <span
              class="content"
            />
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wizard"
    >
      <ul>
        <li
          class="selected"
        >
          <a
            href="#"
          >
            my first step
          </a>
          <span
            class="content"
          />
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Wizard.Button /> renders the button disabled 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-label="Next button"
        class="pf-c-button pf-m-secondary pf-m-disabled wizard-button"
        disabled=""
        type="button"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          height="1em"
          role="img"
          style="vertical-align: -0.125em;"
          viewBox="0 0 512 512"
          width="1em"
        >
          <path
            d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zM140 300h116v70.9c0 10.7 13 16.1 20.5 8.5l114.3-114.9c4.7-4.7 4.7-12.2 0-16.9l-114.3-115c-7.6-7.6-20.5-2.2-20.5 8.5V212H140c-6.6 0-12 5.4-12 12v64c0 6.6 5.4 12 12 12z"
            transform=""
          />
        </svg>
         
        Next
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-label="Next button"
      class="pf-c-button pf-m-secondary pf-m-disabled wizard-button"
      disabled=""
      type="button"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 512 512"
        width="1em"
      >
        <path
          d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zM140 300h116v70.9c0 10.7 13 16.1 20.5 8.5l114.3-114.9c4.7-4.7 4.7-12.2 0-16.9l-114.3-115c-7.6-7.6-20.5-2.2-20.5 8.5V212H140c-6.6 0-12 5.4-12 12v64c0 6.6 5.4 12 12 12z"
          transform=""
        />
      </svg>
       
      Next
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Wizard.Button /> renders the button with type launch 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-label="Next button"
        class="pf-c-button pf-m-secondary wizard-button"
        type="button"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          height="1em"
          role="img"
          style="vertical-align: -0.125em;"
          viewBox="0 0 512 512"
          width="1em"
        >
          <path
            d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zM140 300h116v70.9c0 10.7 13 16.1 20.5 8.5l114.3-114.9c4.7-4.7 4.7-12.2 0-16.9l-114.3-115c-7.6-7.6-20.5-2.2-20.5 8.5V212H140c-6.6 0-12 5.4-12 12v64c0 6.6 5.4 12 12 12z"
            transform=""
          />
        </svg>
         
        Next
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-label="Next button"
      class="pf-c-button pf-m-secondary wizard-button"
      type="button"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 512 512"
        width="1em"
      >
        <path
          d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zM140 300h116v70.9c0 10.7 13 16.1 20.5 8.5l114.3-114.9c4.7-4.7 4.7-12.2 0-16.9l-114.3-115c-7.6-7.6-20.5-2.2-20.5 8.5V212H140c-6.6 0-12 5.4-12 12v64c0 6.6 5.4 12 12 12z"
          transform=""
        />
      </svg>
       
      Next
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Wizard.Button /> renders the button with type next 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-label="Go to next button"
        class="pf-c-button pf-m-secondary wizard-button"
        type="button"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          height="1em"
          role="img"
          style="vertical-align: -0.125em;"
          viewBox="0 0 512 512"
          width="1em"
        >
          <path
            d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zM140 300h116v70.9c0 10.7 13 16.1 20.5 8.5l114.3-114.9c4.7-4.7 4.7-12.2 0-16.9l-114.3-115c-7.6-7.6-20.5-2.2-20.5 8.5V212H140c-6.6 0-12 5.4-12 12v64c0 6.6 5.4 12 12 12z"
            transform=""
          />
        </svg>
         
        Go to next
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-label="Go to next button"
      class="pf-c-button pf-m-secondary wizard-button"
      type="button"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 512 512"
        width="1em"
      >
        <path
          d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zM140 300h116v70.9c0 10.7 13 16.1 20.5 8.5l114.3-114.9c4.7-4.7 4.7-12.2 0-16.9l-114.3-115c-7.6-7.6-20.5-2.2-20.5 8.5V212H140c-6.6 0-12 5.4-12 12v64c0 6.6 5.4 12 12 12z"
          transform=""
        />
      </svg>
       
      Go to next
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Wizard.Step /> renders the step as completed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="completed"
      >
        <a
          href="#"
        >
          my first step
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="color: rgb(92, 184, 92);"
            viewBox="0 0 512 512"
            width="1em"
          >
            <path
              d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 48c110.532 0 200 89.451 200 200 0 110.532-89.451 200-200 200-110.532 0-200-89.451-200-200 0-110.532 89.451-200 200-200m140.204 130.267l-22.536-22.718c-4.667-4.705-12.265-4.736-16.97-.068L215.346 303.697l-59.792-60.277c-4.667-4.705-12.265-4.736-16.97-.069l-22.719 22.536c-4.705 4.667-4.736 12.265-.068 16.971l90.781 91.516c4.667 4.705 12.265 4.736 16.97.068l172.589-171.204c4.704-4.668 4.734-12.266.067-16.971z"
              transform=""
            />
          </svg>
        </a>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="completed"
    >
      <a
        href="#"
      >
        my first step
        <svg
          aria-hidden="true"
          fill="currentColor"
          height="1em"
          role="img"
          style="color: rgb(92, 184, 92);"
          viewBox="0 0 512 512"
          width="1em"
        >
          <path
            d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 48c110.532 0 200 89.451 200 200 0 110.532-89.451 200-200 200-110.532 0-200-89.451-200-200 0-110.532 89.451-200 200-200m140.204 130.267l-22.536-22.718c-4.667-4.705-12.265-4.736-16.97-.068L215.346 303.697l-59.792-60.277c-4.667-4.705-12.265-4.736-16.97-.069l-22.719 22.536c-4.705 4.667-4.736 12.265-.068 16.971l90.781 91.516c4.667 4.705 12.265 4.736 16.97.068l172.589-171.204c4.704-4.668 4.734-12.266.067-16.971z"
            transform=""
          />
        </svg>
      </a>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Wizard.Step /> renders the step as completed and locked 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="locked"
      >
        <a
          href="#"
        >
          my first step
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="color: rgb(189, 194, 206);"
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
              transform=""
            />
          </svg>
        </a>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="locked"
    >
      <a
        href="#"
      >
        my first step
        <svg
          aria-hidden="true"
          fill="currentColor"
          height="1em"
          role="img"
          style="color: rgb(189, 194, 206);"
          viewBox="0 0 448 512"
          width="1em"
        >
          <path
            d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
            transform=""
          />
        </svg>
      </a>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
